### ISSUE TITLE
redundans appears to finish but assembly unchanged

### ISSUE THREAD

**Issue by jmodlis** [Gabaldonlab/redundans#53]
Hi,  I've tried running redundant on several assemblies - when I only use the assembly and Pacbio data, redundans runs but the final assembly is no different from the input assembly.

I have tried it with some Illumina data and it did work, the test worked as well. 

Below is the log file (the first line just has the version printed)
```
.14a
Options: Namespace(fasta='/data/omicscore/20180903/rawData/job1128_asm2.fasta', fastq=[], identity=0.51, iters=2, joins=5, limit=0.2, linkratio=0.7, log=<open file '<stderr>', mode 'w' at 0x7fa85608a1e0>, longreads=['/data/omicscore/20180903/rawData/m54089_180805_144407.subreads.fastq.gz'], mapq=10, mem=16, minLength=200, nocleaning=True, nogapclosing=True, norearrangements=False, noreduction=True, noscaffolding=True, outdir='/data/omicscore/20180903/processedData/redundans/hgap4', overlap=0.8, reference='', resume=False, threads=4, tmp='/data/omicscore/20180903/processedData/redundans/tmp1', usebwa=False, verbose=True)

##################################################
[Wed Sep 26 15:24:27 2018] Reduction...
#file name	genome size	contigs	heterozygous size	[%]	heterozygous contigs	[%]	identity [%]	possible joins	homozygous size	[%]	homozygous contigs	[%]
lastdb: I was installed here with multi-threading disabled
lastal: can't open file: /data/omicscore/20180903/processedData/redundans/hgap4/contigs.fa.prj
/nfs/software/helmod/apps/Core/Anaconda/2.5.0-fasrc01/x/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
[WARNING] Nothing reduced!
/data/omicscore/20180903/processedData/redundans/hgap4/contigs.fa	101217907	169	0	0.00	0	0.00	0.000	0	101217907	100.00	169	100.00

##################################################
[Wed Sep 26 15:24:51 2018] Scaffolding with long reads...
 iteration 1...
lastal: I was installed here with multi-threading disabled

##################################################
[Wed Sep 26 15:25:52 2018] Final reduction...
#file name	genome size	contigs	heterozygous size	[%]	heterozygous contigs	[%]	identity [%]	possible joins	homozygous size	[%]	homozygous contigs	[%]
lastdb: I was installed here with multi-threading disabled
lastal: can't open file: /data/omicscore/20180903/processedData/redundans/hgap4/scaffolds.longreads.fa.prj
[WARNING] Nothing reduced!
/data/omicscore/20180903/processedData/redundans/hgap4/scaffolds.longreads.fa	101217907	169	0	0.00	0	0.00	0.000	0	101217907	100.00	169	100.00

##################################################
[Wed Sep 26 15:26:07 2018] Reporting statistics...
#fname	contigs	bases	GC [%]	contigs >1kb	bases in contigs >1kb	N50	N90	Ns	longest
/data/omicscore/20180903/rawData/job1128_asm2.fasta	169	101217907	35.466	169	101217907	1147021	339180	0	4668767
/data/omicscore/20180903/processedData/redundans/hgap4/contigs.fa	169	101217907	35.466	169	101217907	1147021	339180	0	4668767
/data/omicscore/20180903/processedData/redundans/hgap4/contigs.reduced.fa	169	101217907	35.466	169	101217907	1147021	339180	0	4668767
/data/omicscore/20180903/processedData/redundans/hgap4/scaffolds.longreads.1.fa	169	101217907	35.466	169	101217907	1147021	339180	0	4668767
/data/omicscore/20180903/processedData/redundans/hgap4/scaffolds.longreads.fa	169	101217907	35.466	169	101217907	1147021	339180	0	4668767
/data/omicscore/20180903/processedData/redundans/hgap4/scaffolds.reduced.fa	169	101217907	35.466	169	101217907	1147021	339180	0	4668767

##################################################
[Wed Sep 26 15:26:22 2018] Cleaning-up...
#Time elapsed: 0:01:55.791153

gzip: stdout: Broken pipe
```

Note that it does appear to exit strangely, but it did this with the illumina assembly as well.

For what it is worth, I am currently running the pyScaf that came with redundans on the same file and it is still running, below is the log file.  I did have to install last and do an export PATH to the last binaries within the shell script that included the pyScaf command.
```
Currently Loaded Modules:
  1) gmp/6.1.1-fasrc02    4) gcc/6.2.0-fasrc01        7) zlib/1.2.8-fasrc07
  2) mpfr/3.1.4-fasrc02   5) Anaconda/2.5.0-fasrc01
  3) mpc/1.0.3-fasrc04    6) python/2.7.11-fasrc01

0.12a
Options: Namespace(dotplot='png', fasta='/data/omicscore/20180903/rawData/job1128_asm2.fasta', fastq=None, identity=0.33, joins=5, linkratio=0.7, load=0.2, log=<open file '<stderr>', mode 'w' at 0x7f1c1cf8b1e0>, longreads=['/data/omicscore/Sherwood-Costa-20180903/rawData/m54089_180805_144407.subreads.fastq.gz'], mapq=10, maxgap=0, nofilter_overlaps=False, norearrangements=False, output=<open file '/data/omicscore/20180903/processedData/pyscaf/scaffolds.fa', mode 'w' at 0x7f1c1cbf5420>, overlap=0.66, ref='', threads=4)
 maxgap cut-off of 1012179 bp

##################################################
[Wed Sep 26 14:51:54 2018][    7 Mb] Aligning long reads on contigs...
```

Thanks for any insight that you may have!
Jen

**Comment by lpryszcz** [comment]
Hi Jen, from errors it seems that there was some issue with your lastal installation. That's why redundans fails on your genomes. Make sure you have the latest version installed and running. 
```
lastdb: I was installed here with multi-threading disabled
lastal: can't open file: 
```

**Comment by jmodlis** [comment]
Hi,  

So I did have some issues with the install of lastal, but before I ran redundans, I did get it installed using this:
```
cd ./redundans/bin/last/ 
make clean
make CXXFLAGS=-O3
```

And it appears to work:
```
[jlm50]$ pwd
/data/omicscore/20180813/scripts/redundans/bin/last/src
[jlm50]$ ./lastdb
./lastdb: please give me an output name and sequence file(s)

Usage: ./lastdb [options] output-name fasta-sequence-file(s)
Prepare sequences for subsequent alignment with lastal.

Main Options:
-h, --help: show all options and their default settings, and exit
-p: interpret the sequences as proteins
-R: repeat-marking options (default=10)
-c: soft-mask lowercase letters (in reference *and* query sequences)
-u: seeding scheme (default: YASS for DNA, else exact-match seeds)

[jlm50]$ ./lastal
./lastal: please give me a database name and sequence file(s)

Usage: ./lastal [options] lastdb-name fasta-sequence-file(s)
Find and align similar sequences.

Cosmetic options:
-h, --help: show all options and their default settings, and exit
-V, --version: show version information, and exit
-v: be verbose: write messages about what lastal is doing
-f: output format: TAB, MAF, BlastTab, BlastTab+ (default=MAF)
```

But this is what I get when I run redundans on Illumina data.  Again, there is no change in the number of contigs.  Is there a better way to check the lastal installation?

Also, could the gzip error have anything to do with this?

```
Currently Loaded Modules:
  1) Anaconda/2.5.0-fasrc01   2) python/2.7.11-fasrc01   3) zlib/1.2.8-fasrc07

0.14a
Options: Namespace(fasta='/data/omicscore/20180813/rawData/assembly/genome_070416.fasta', fastq=[], identity=0.51, iters=2, joins=5, limit=0.2, linkratio=0.7, log=<open file '<stderr>', mode 'w' at 0x7f3b487411e0>, longreads=['/data/omicscore/20180813/processedData/AllSkip.fastq.gz'], mapq=10, mem=16, minLength=200, nocleaning=True, nogapclosing=True, norearrangements=False, noreduction=False, noscaffolding=True, outdir='/data/omicscore/20180813/processedData/redundans_newNanopore', overlap=0.8, reference='', resume=False, threads=4, tmp='/data/omicscore/20180813/processedData/redundans_newNanopore/tmp', usebwa=False, verbose=True)

##################################################
[Wed Dec  5 12:40:27 2018] Scaffolding with long reads...
 iteration 1...
lastal: I was installed here with multi-threading disabled

##################################################
[Wed Dec  5 12:47:30 2018] Reporting statistics...
#fname  contigs bases   GC [%]  contigs >1kb    bases in contigs >1kb   N50     N90     Ns      longest
/data/omicscore/20180813/rawData/assembly/genome_070416.fasta  2065    431199575       35.784  2065    431199575       550413  99260   42      4769731
/data/omicscore/20180813/processedData/redundans_newNanopore/contigs.fa 2065    431199575       35.784  2065    431199575       550413  99260   42      4769731
/data/omicscore/20180813/processedData/redundans_newNanopore/scaffolds.longreads.1.fa   2065    431199575       35.784  2065    431199575       550413  99260   42      4769731
/data/omicscore/20180813/processedData/redundans_newNanopore/scaffolds.longreads.fa     2065    431199575       35.784  2065    431199575       550413  99260   42      4769731

##################################################
[Wed Dec  5 12:48:40 2018] Cleaning-up...
#Time elapsed: 0:08:13.025806

gzip: stdout: Broken pipe
~
```

Thanks,
Jen

**Comment by lpryszcz** [comment]
Make sure your version of LASTal is at least lastal 852 and with multi-threading enabled, as that's what it's complaining rigth now...
```redundans/bin/last/src/lastal --version```

**Comment by jmodlis** [comment]
Hi,

I was able to get it working.  I'm working on a cluster and had to load gcc 6.2 and zlib 1.2.8 to install last correctly (multi-threading was indeed the issue). gcc/g++ 4.6 is the default compiler, which seemed to be a requirement of SNAP, so I used 4.6 to install SNAP (I tried various other gcc versions but only 4.6 seemed to work).  To run the program I then load the gcc 6.2 and zlib 1.2.8 modules.

Thanks,
Jen
